packages/liveblocks-react-lexical/src/comments/floating-threads.tsx

```diff
@@ -0,0 +1,321 @@
+  range: Range;
+  container: HTMLElement;
+  children: ReactNode;
+}
+
+export const FLOATING_THREAD_COLLISION_PADDING = 10;
+
+function FloatingThreadPortal(props: FloatingThreadPortalProps) {
+  const { container, range, children, className, style, ...divProps } = props;
+
+  const {
+    refs: { setReference, setFloating },
+    strategy,
+    x,
+    y,
+  } = useFloating({
+    strategy: "absolute",
+    placement: "bottom",
+    middleware: [
+      flip({ padding: FLOATING_THREAD_COLLISION_PADDING, crossAxis: false }),
+      offset(10),
+      hide({ padding: FLOATING_THREAD_COLLISION_PADDING }),
+      shift({
+        padding: FLOATING_THREAD_COLLISION_PADDING,
+        limiter: limitShift(),
+      }),
+      size({
+        padding: FLOATING_THREAD_COLLISION_PADDING,
+        apply({ availableWidth, availableHeight, elements }) {
+          elements.floating.style.setProperty(
+            "--lb-lexical-floating-threads-available-width",
+            `${availableWidth}px`
+          );
+          elements.floating.style.setProperty(
+            "--lb-lexical-floating-threads-available-height",
+            `${availableHeight}px`
+          );
+        },
+      }),
+    ],
+    whileElementsMounted: (...args) => {
+      return autoUpdate(...args, {
+        animationFrame: true,
+      });
+    },
+  });
+
+  useLayoutEffect(() => {
+    setReference({
+      getBoundingClientRect: () => range.getBoundingClientRect(),
+    });
+  }, [setReference, range]);
+
+  return createPortal(
+    <div
+      ref={setFloating}
+      {...divProps}
+      style={{
+        ...style,
+        position: strategy,
+        top: 0,
+        left: 0,
+        transform: `translate3d(${Math.round(x)}px, ${Math.round(y)}px, 0)`,
+        minWidth: "max-content",
+      }}
+      className={classNames(
+        "lb-root lb-portal lb-elevation lb-lexical-floating-threads",
+        className
+      )}
+    >
+      {children}
+    </div>,
+    container
+  );
+}
+
+interface ThreadWrapperProps extends ThreadProps {
+  thread: ThreadData;
+  Thread: ComponentType<ThreadProps>;
+  onEscapeKeydown: () => void;
+}
+
+function ThreadWrapper({
+  thread,
+  Thread,
+  style,
+  onEscapeKeydown,
+  onKeyDown,
+  ...threadProps
+}: ThreadWrapperProps) {
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      onKeyDown?.(event);
+
+      // TODO: Add ability to preventDefault on keydown to override the default behavior, e.g. to show an alert dialog
+      if (event.key === "Escape") {
+        onEscapeKeydown();
+      }
+    },
+    [onEscapeKeydown, onKeyDown]
+  );
+
+  return (
+    <Thread
+      thread={thread}
+      onKeyDown={handleKeyDown}
+      style={{
+        maxHeight: "var(--lb-lexical-floating-threads-available-height)",
+        maxWidth: "var(--lb-lexical-floating-threads-available-width)",
+        overflow: "auto",
+        ...style,
+      }}
```

Comment:
Added the `maxHeight` and `maxWidth` style here on the `Thread` component within `ThreadWrapper` component.